### PR TITLE
[DeepSeekV4] Add flash_mla for CSA forward

### PR DESCRIPTION
**Pull request by lshpku**
### PR Category
Operator Mechanism

### PR Types
New features

### Description
新增 [FlashMLA](https://github.com/PFCCLab/FlashMLA) 库，使用 flash_mla_sparse_fwd kernel 替换原有的 tilelang 前向实现（反向见 https://github.com/PaddlePaddle/PaddleFleet/pull/1101 ）

<b>使用方法：</b>
* 通过环境变量开启：`export USE_FLASH_MLA=1`; `export USE_CUDNN_DSA=1`
* <b>注：</b>以上两个开关需要同时开启，因为 flash_mla 和 cudnn DSA 是配套的，它们的 lse 和原来的 tilelang 的定义不同，只开前向或者反向会导致 lse 值域错误，loss 起飞

* 以上环境变量生效的前提是开启了 tilelang，参考 json 配置：
```json
  "csa_tilelang_backend": "attention_paddle_compat",
  "csa_tilelang_enable_indexer": true,
  "csa_tilelang_enable_sparse_attn": true,
```

### 是否引起精度变化
是

### CODE REVIEW

File: src/paddlefleet/tilelang_ops/attn/sparse_mqa.py [hunk 1]
```diff
@@ -12,10 +12,24 @@
...
 import paddle
+import paddle.nn.functional as F
+from paddle.utils import strtobool
 
 from . import sparse_mqa_fwd
 
+_USE_FLASH_MLA = strtobool(os.getenv("USE_FLASH_MLA", "0"))
+
+if _USE_FLASH_MLA:
+    try:
```
[the remark this answers is not in the document]
Comment (lshpku) [reply src/paddlefleet/tilelang_ops/attn/sparse_mqa.py:26]: 不建议修复
